greeting/build.gradle.kts

```
@@ -0,0 +1,94 @@
+import org.jetbrains.kotlin.gradle.plugin.mpp.KotlinNativeTarget
+
+plugins {
+    id( "com.android.library")
+    id("org.jetbrains.kotlin.multiplatform")
+}
+
+group = "org.greeting"
+version = 1.0
+
+android {
+    compileSdkVersion(27)
+    defaultConfig {
+        minSdkVersion(15)
+    }
+    buildTypes {
+        getByName("release") {
+            isMinifyEnabled = true
+            proguardFiles(getDefaultProguardFile("proguard-android-optimize.txt"), "proguard-rules.pro")
+        }
+    }
+}
+
+dependencies {
+    // Specify Kotlin/JVM stdlib dependency.
+    implementation("org.jetbrains.kotlin:kotlin-stdlib-jdk7")
+
+    testImplementation("junit:junit:4.12")
+    testImplementation("org.jetbrains.kotlin:kotlin-test")
+    testImplementation("org.jetbrains.kotlin:kotlin-test-junit")
+
+    androidTestImplementation("junit:junit:4.12")
+    androidTestImplementation("org.jetbrains.kotlin:kotlin-test")
+    androidTestImplementation("org.jetbrains.kotlin:kotlin-test-junit")
+}
+
+kotlin {
+   android("androidLib")
+   
+   val buildForDevice = project.findProperty("device") as? Boolean ?: false
+   val iosTarget = if(buildForDevice) iosArm64("ios") else iosX64("ios")
+   iosTarget.binaries {
+      framework {
+         // Disable bitcode embedding for the simulator build.
+         if (!buildForDevice) {
+            embedBitcode("disable")
+         }
+      }
+   }
+
+   sourceSets {
+      commonMain {
+         dependencies {
+            implementation( "org.jetbrains.kotlin:kotlin-stdlib-common")
+         }
+      }
+      commonTest {
+         dependencies {
+            implementation( "org.jetbrains.kotlin:kotlin-test-common")
+             implementation("org.jetbrains.kotlin:kotlin-test-annotations-common")
+         }
+      }
+    }
+}
+
+tasks.register("copyFramework") {
+    val buildType = project.findProperty("kotlin.build.type") as? String ?: "DEBUG"
+    dependsOn("link${buildType.toLowerCase().capitalize()}FrameworkIos")
+
+    doLast {
+        val srcFile = (kotlin.targets.named("ios") as KotlinNativeTarget).binaries.getFramework(buildType).outputFile
```

The original code fragment fails with the following error:
```
> org.gradle.api.internal.DefaultNamedDomainObjectCollection$ExistingNamedDomainObjectProvider_Decorated cannot be cast to org.jetbrains.kotlin.gradle.plugin.mpp.KotlinNativeTarget
```
because `named` returns a `NamedDomainObjectProvider` instead of an object stored in the container. The following change should fix this problem:


```suggestion
        val srcFile = (kotlin.targets["ios"] as KotlinNativeTarget).binaries.getFramework(buildType).outputFile
```

---

Fixed here https://github.com/Kotlin/mpp-example/pull/31/commits/b04390244185c1df0702521b822d5163ee94cf62